validate.go

```diff
@@ -18,34 +19,127 @@ func validate(payload []byte) ([]byte, error) {
 
 	logger.Info("validating request")
 
-	data := gjson.GetBytes(
+	if len(settings.AllowedHostPaths) == 0 {
+		// empty settings, accepting
+		return kubewarden.AcceptRequest()
+	}
+
+	volumes := gjson.GetBytes(
 		payload,
-		"request.object.metadata.name")
+		"request.object.spec.volumes")
+	if !volumes.Exists() {
+		// pod defines no volumes, accepting
+		return kubewarden.AcceptRequest()
+	}
+
+	var volumeMounts gjson.Result
+	// workaround bug when obtaining array
+	// request.object.spec.containers.volumeMounts
+	for _, object := range gjson.GetBytes(payload,
+		"request.object.spec.containers").Array() {
+		if gjson.Get(object.String(), "volumeMounts").Exists() {
+			volumeMounts = gjson.Get(object.String(), "volumeMounts")
+		}
+	}
 
-	if !data.Exists() {
-		logger.Warn("cannot read object name from metadata: accepting request")
+	if !volumeMounts.Exists() {
+		// pod defines no mounts, accepting
 		return kubewarden.AcceptRequest()
 	}
```

Comment:
I find something odd in this logic. Early accepting taking into account a variable that is overriden and that will take the last element only is a bit weak I think.

In general, I think that policies are safer with early rejections, and accepting a minimum amount of times, so is easier to understand what paths lead to acceptance.

I could try to bypass the policy by trying to make the last container valid and safe, but have previous containers mount paths that would otherwise be rejected by the real loop that comes after.

Maybe it's just me though :)

Reply:
Ouch, I think Rafa is right here. I think the right code could not work when a Pod is defining multiple containers.

You should add a test case for that too

Reply:
Good points, thanks! I totally didn't consider multiple containers nor `initContainer`, silly me.

Reply:
Done in https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/ead60048e6a00223a677051e4c2026e2f43302c1, with tests in https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/4faa364cdb9d9f522a8ae688a47188976c1e6337 and https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/5333d9074e44da62bbdaabf7817ad8df4f54ebe4.